xdr3/decode.go

```diff
@@ -428,7 +431,14 @@ func (d *Decoder) decodeFixedArray(v reflect.Value, ignoreOpaque bool) (int, err
 	// Treat [#]byte (byte is alias for uint8) as opaque data unless
 	// ignored.
 	if !ignoreOpaque && v.Type().Elem().Kind() == reflect.Uint8 {
-		data, n, err := d.DecodeFixedOpaque(int32(v.Len()))
+		if v.CanAddr() {
+			/// decode in-place if the array is addressable
+			// (can't obtain a slice from an unaddressable array)
```

Comment:
❔ What's an example of an unaddressable array? I didn't realize they could be unaddressable. I'm also surprised we can copy into an unaddressable array further down. TIL.

Reply:
An example is an array returned by a function (as opposed to a declared variable).

However, after adding a test you are right about not being able to copy to an un-addressable array.

Also, I can't produce an example which passes an unaddressable array to `xdr.Unmarshal()`  (since it checks for a pointer argument) just to `decodeFixedArray` which is internal.

Thus, I will just remove the code for unaddressable arrays (it would had failed before as well due to the `Copy()` anyways)